tests/test_page_objects.py

```diff
@@ -0,0 +1,64 @@
+from datetime import datetime
+
+import attrs
+import pytest
+from web_poet import HttpResponse, ResponseUrl, field
+
+from zyte_common_items import ProductListPage, ProductPage
+
+
+@pytest.mark.parametrize(
+    "page_class",
+    (
+        ProductPage,
+        ProductListPage,
+    ),
+)
+def test_default(page_class):
+    datetime_before = datetime.utcnow().replace(microsecond=0)
+
+    page = page_class(url="https://example.com")
+
+    assert page.metadata.probability is None
+    assert page.url == "https://example.com"
+
+    page_datetime_string = page.metadata.dateDownloaded
+    assert page_datetime_string.endswith("Z")
+    page_datetime = datetime.fromisoformat(page_datetime_string[:-1])
+    datetime_after = datetime.utcnow().replace(microsecond=0)
+    assert datetime_before <= page_datetime <= datetime_after
+
+
+@pytest.mark.asyncio
+async def test_example():
+    datetime_before = datetime.utcnow().replace(microsecond=0)
+
+    @attrs.define
+    class BookPage(ProductPage):
+        response: HttpResponse
```

Comment:
It occurred to me that majority of the user's use case would always need the `HttpResponse` dependency.

I'm thinking if we should have `ProductPage` inherit from `web_poet.WebPage` instead of `web_poet.ItemPage` to already include `HttpResponse` from the get go. This could save the user one line of code like this one.

An issue with `web_poet.WebPage` though is that it doesn't mark such dependency to be `Optional[HttpResponse]`. So in cases where users don't need the response, *mypy* would complain if they pass `None`. However, they could simply pass a `HttpResponse` with some fake values to quiet *mypy* down. In any case, I think this would be an edge case since I expect users to be always using the response via `self.css()` and `self.xpath()` inside the PO.

Thoughts @Gallaecio @kmike ?

Reply:
If a page object uses browser HTML, that could be a problem, as you cannot remove attrs attributes, i.e. both response body and browser HTML would be provided through dependency injection.

Reply:
I think both points are valid. It seems we need to have both classes: with and without HttpResponse defined as a dependency. The main question is where we should have them. 

We should definitely provide a class without HttpResponse dependency in this repo. 

For a ProductPage with HttpResponse dependency there are 2 options:

1) Add another class here. Probably rename ProductPage to BaseProductPage, and add ProductPage with response dependency.
2) Encourage users to subclass ProductPage defined here, not use it directly.

Reply:
Isn’t 2 the current scenario? Don’t users need to subclass ProductPage either way to define additional fields?

Reply:
Good point about having `browserHtml` as it could possibly be the 2nd most common use case.

Option 1 with `BaseProductPage` and `ProductPage` sounds good to me. 👍

Reply:
> Isn’t 2 the current scenario?

I mean that we might need to advice users to create their own base ProductPage, which is a subclass of zyte_common_items.ProductPage, with the response: HttpResponse defined.

Reply:
[Went for option 1](https://github.com/zytedata/zyte-common-items/pull/19/commits/a14a9bdc809c5263b6ff95e70696c3e20dbba3ed).